### DISCUSSION TITLE
On a SQL join get only the value and not an object with key and value in query written in JS

### DISCUSSION THREAD

**Discussion by foufrix** [supabase/supabase#18531] [Questions] [answered]
I have a table `texts` which is :
| id         | en     | fr |
|--------------|-----------|------------|
| appleDescription | It's an apple      | C'est une pomme   |
| introTitle     | Hi everyone nice to me you  | Bonjour tout le monde ravie de vous rencontrer      |

I refer id to another table
Currently I do this request and it give me as results:
```
const query = supabase.from('activities').select('id, icon, introTitle (fr)');
console.log(query) //activities:  [{"id":"1","icon":"gate","introTitle":{"fr":"Bonjour tout le monde ravie de vous rencontrer"}}]
```

How can I get :

```JSON
[{"id":"1","icon":"gate","introTitle": "Bonjour tout le monde ravie de vous rencontrer"}]
```

Search around for hours about that, and saw some recommendations about using `->` but did not manage to have the proper syntax to make it works

Thanks in advance

**Comment by GaryAustin1** [comment]
See spread operator here:
https://supabase.com/blog/postgrest-11-prerelease
It will get you closer by dropping the object structure, but still not quite what you show.
You can always post process your result in the client or do a view where you have a french view and a english view.

**Reply by foufrix** [reply]
Dang, found that crazy because we can manipulate a JSON object directly with https://supabase.com/docs/guides/database/json?database-method=js&language=js#query-the-jsonb-data, and it's kind of the same thing.
No way to get the results and then trim them directly in the request without having to create a custom mapper in the front? :(

Though we can do that with a custom SQL request and request this endpoint, seems better.

**Reply by GaryAustin1** [reply, marked as answer]
I don't know of one, but sometimes PostgREST does things I've not encountered.  I think a view might be the better way to go.

**Reply by foufrix** [reply]
Thanks, we'll go with the view then
Have a good day :)

**Reply by GaryAustin1** [reply]
No a view uses your existing tables but formats the data as you want.
This is just a quick thing from Supabase on it:  https://supabase.com/docs/guides/database/tables#views   
There would be lots of info on the web for Postgres views.

**Reply by foufrix** [reply]
yep saw that, sorry deleted my old message
I see that I cannot link a view as I did with a table
How would you do that in a proper way? Struggle to see the efficient logic to apply here.
I can create a view with 

id | en 
-- | -- 
appleDescription | It's an apple 
introTitle | Hi everyone nice to me you

But don't see how it solves the problem, as I cannot create a relation now
`const query = supabase.from('activities').select('id, icon, introTitle(fr)');`

**Reply by GaryAustin1** [reply]
You would do the linking in the view.  So one view would be your activities table with a column introTitle joined from the other table in English.  The other view would be the activities table with a column introTitle joined with the French column.

There should be info on doing joins in the example and plenty online. 

You basically are creating two virtual windows into the activities table joined with the introTitle table. 

Sorry, I probably can't be more help with it as I don't use views myself very often, but seems ideal for what you are trying to do.

**Reply by foufrix** [reply]
I understood, thanks for the help, had the wrong thinking
I thought I needed a `texts` view with `fr` and `en`
But no, I need to create a view of activities for both languages. This means I will end up with many view tables as `activities` is one of many table I have :(
I'll see if finding a universal algorithm to map the frontend response is not better then
